### PR TITLE
Make sure we have addresses for newly discovered nodes

### PR DESCRIPTION
**Pull request by justinsb**
Partial workaround for #6072, I think, although I'm not sure whether fixing this is in scope for #6072.  Right now, NodeAddresses is _only_ called by nodecontroller.go, and only if the instances exist when the node-cloud-controller is first started.

Hopefully we can rationalize all this code once it becomes clear who is responsible for the node status data.

### CODE REVIEW

File: pkg/cloudprovider/controller/nodecontroller.go [hunk 1]
```diff
@@ -223,6 +223,16 @@ func (nc *NodeController) SyncCloudNodes() error {
 	// Create nodes which have been created in cloud, but not in kubernetes cluster.
 	for _, node := range matches.Items {
 		if _, ok := nodeMap[node.Name]; !ok {
+			glog.Infof("Querying addresses for new node: %s", node.Name)
```
Comment (vmarmol) [inline pkg/cloudprovider/controller/nodecontroller.go]: We should probably lower this to V(2)?

**Comment by justinsb** [comment]
Dropped to V(3).  The whole when we do/don't query NodeAddresses is a little messy right now (#6072), but when this gets fixed we definitely don't want this log message hanging around at v(0), so I took your suggestion - thanks!

**Comment by vmarmol** [comment]
Assigning to @gmarek

**Comment by vmarmol** [comment]
You can probably squash the commit too :) This LGTM, but I want @gmarek to take a look since he has more context.

**Comment by justinsb** [comment]
Squashed!  Definitely agree with getting a second opinion here; it's not clear to me how this works at all if the minions come up after the master.

File: pkg/cloudprovider/controller/nodecontroller.go [hunk 2]
```diff
@@ -223,6 +223,16 @@ func (nc *NodeController) SyncCloudNodes() error {
 	// Create nodes which have been created in cloud, but not in kubernetes cluster.
 	for _, node := range matches.Items {
 		if _, ok := nodeMap[node.Name]; !ok {
+			glog.V(3).Infof("Querying addresses for new node: %s", node.Name)
+			nodeList := &api.NodeList{}
+			nodeList.Items = []api.Node{node}
+			_, err = nc.PopulateAddresses(nodeList)
+			if err != nil {
```
Comment (gmarek) [inline pkg/cloudprovider/controller/nodecontroller.go]: Please add a comment saying that Nodes for which we fail populating address are skipped.

**Comment by gmarek** [comment]
Please add a test for this behaviour, so we won't regress in future.

**Comment by justinsb** [comment]
Added unit test

**Comment by gmarek** [comment]
There are some test failures now - the same on travis and shippable. They seem unrelated, but please make sure of that.

**Comment by vmarmol** [comment]
Now green! Merging.

**Pull request merged**